explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -44,179 +42,150 @@
-  }
-
-  @Test
-  public void get_CertificateNotFoundTwice_ShouldThrowExplorerException() {
-    // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseCertificateNotFound);
+    when(clientService.executeContract(eq( HOLDER_ID + "GET"), any(JsonObject.class)))
+        .thenThrow(clientExceptionCertificateNotFound);
+    doNothing().when(clientService).registerCertificate();
 
     // Act-assert
     assertThatThrownBy(() -> explorer.get(ASSET_ID))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+        .hasMessage("403 Certificate not found");
   }
 
   @Test
-  public void get_ContractNotFoundTwice_ShouldThrowExplorerException() {
+  public void get_CertificateNotFound_ShouldThrowExplorerException() {
     // Arrange
     when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseContractNotFound, contractExecutionResponseContractNotFound);
+        .thenThrow(clientExceptionCertificateNotFound);
 
     // Act-assert
     assertThatThrownBy(() -> explorer.get(ASSET_ID))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("404 contract not found");
+        .hasMessage("403 Certificate not found");
   }
 
   @Test
-  public void scan_ThrowExceptionOnlyOneTime_ShouldReturnSuccess() {
+  public void scan_ShouldReturnSuccess() {
     // Arrange
-    when(contractExecutionResponseOK.getResult())
-        .thenReturn(
-            "{\"result\":\"success\",\"history\":[{\"id\":\"bar\",\"age\":0,\"data\":{}}]}");
+    JsonObject expected = Json.createReader(
+            new StringReader("{\"status\":\"OK\"," +
+                    "\"history\":[{\"id\":\"bar\",\"age\":0,\"data\":{}}]}"))
+            .readObject();
+    when(contractExecutionResultOK.getResult()).thenReturn(Optional.of(expected));
     when(clientService.executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseContractNotFound,
-            contractExecutionResponseOK);
+        .thenReturn(contractExecutionResultOK);
 
     // Act
-    JsonArray result = explorer.scan(ASSET_ID, Json.createObjectBuilder().build());
+    JsonArray actual = explorer.scan(ASSET_ID, Json.createObjectBuilder().build());
 
     // Assert
-    verify(clientService, Mockito.times(3))
-        .executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class));
-    assertThat(result.getJsonObject(0).getString("id")).isEqualTo("bar");
-    assertThat(result.getJsonObject(0).getInt("age")).isEqualTo(0);
+    verify(clientService).executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class));
+    assertThat(actual.getJsonObject(0).getString("id")).isEqualTo("bar");
```

Comment:
Can we just do something like this ?
```
assertThat(actual.getJsonObject(0)).isEqualTo(expected);
```

Reply:
Thank you for the review comment.
Review changes updated in 877c01f